packages/txservice/src/provider.ts

```diff
@@ -450,4 +464,85 @@ export class ChainRpcProvider {
   private incrementNonce() {
     this._nonce++;
   }
+
+  private async monitorBlockades() {
+    while (true) {
+      // First, check to see that there are pending txs
+      if (Object.keys(this.pendingTxs).length === 0) {
+        return;
+      }
+
+      // Get the blocking nonce and corresponding record
+      const blockade = await this.provider.getTransactionCount("pending");
+
+      // If the blockade is our *current* nonce, we can ignore since it will
+      // be sent at the next tx
+      // TODO: ^^ is this a safe assumption
+      if (blockade === (await this.getNonce())) {
+        return;
+      }
+
+      const { added, response } = this.pendingTxs[blockade] ?? {};
+
+      if (!added && !response) {
+        // This means we have no record at the blocking nonce *but* we do
+        // have pending transactions that should get mined.
+
+        // Send a dummy transaction at the blockcade nonce to get the queue
+        // moving
+        this.logger.info({ blockade, nonce: await this.getNonce() }, "Sending dummy tx to unblock");
+        // TODO: this should be sent at the highest gas to clear the queue
+        // quickly. ANd gas limit should be fixed
+        const response = await this.sendTransaction({
+          to: constants.AddressZero,
+          value: BigNumber.from(0),
+          nonce: blockade,
+          data: "0x",
+          gasLimit: BigNumber.from(10),
+          gasPrice: BigNumber.from("0xmaxgas"),
+          chainId: this.chainId,
+        });
+        if (response.isErr()) {
+          // TODO: handle error
+          return;
+        }
+        // NOTE: this may not be the best way to confirm this tx. the point is
+        // it should *not* be bumped, it should be dealt with
+        // here and immediately to free up the queue of pending txs
+        await this.confirmTransaction(response.value, 2);
+        return;
+      }
+
+      // Now you have the transaction response that you need to clear.
+      // Resend this transaction at a high gas price.
+
+      // Ideally, how high this gas price is should be determined by how
+      // long youve been waiting, and how many other txs in the queue.
+
+      const elapsed = Date.now() - added;
+      const TOO_LONG = 2 * 60 * 1000; // should be config determined, for now 2min
+
+      const blocked = Object.keys(this.pendingTxs).filter((nonce) => parseInt(nonce) > blockade).length;
+
+      const gasPrice =
+        elapsed >= TOO_LONG || blocked > 10
+          ? BigNumber.from("0xmaxgas")
+          : response.gasPrice!.add(BigNumber.from("0xgasbump").mul(2));
+
+      const resubmitted = await this.sendTransaction({
```

Comment:
This def creates an antipattern with what's already in place with txservice, but I totally agree with this logic surface level